platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -1230,11 +1210,15 @@ private void writeLeavesToPathToKeyValue(
 
         // iterate over leaf records to delete
         deletedLeaves.forEach(leafRecord -> {
+            final long path = leafRecord.getPath();
             // update objectKeyToPath
             if (isLongKeyMode) {
-                longKeyToPath.put(((VirtualLongKey) leafRecord.getKey()).getKeyAsLong(), INVALID_PATH);
+                final long key = ((VirtualLongKey) leafRecord.getKey()).getKeyAsLong();
+                //                longKeyToPath.put(((VirtualLongKey) leafRecord.getKey()).getKeyAsLong(),
+                // INVALID_PATH);
+                longKeyToPath.putIfEqual(key, path, INVALID_PATH);
             } else {
-                objectKeyToPath.delete(leafRecord.getKey());
+                objectKeyToPath.deleteIfEquals(leafRecord.getKey(), path);
```

Comment:
nit: `putIfEqual` but `deleteIfEquals`. Could you please make sure that the naming is consistent?

Reply:
This `putIfEqual()` existed before this change, it's a part of `CASableLongIndex` interface. It makes sense to introduce the new method with the same name for consistency. I'll change everything to `putIfEqual()` and `deleteIfEqual()`